tests/testcases.py

```diff
@@ -49,12 +50,14 @@ def setUp(self):
         self.client_output = []
         self.capture_handler = TestCaptureHandler(self.client_output)
         logging.getLogger(constants.SOCKET_LOGGER_NAME).addHandler(self.capture_handler)
+        reset_memoize_cache()
 
     def tearDown(self):
         os.remove(self.temp_config_path)
         shutil.rmtree(self.temp_specs_path)
         shutil.rmtree(self.temp_repos_path)
         logging.getLogger(constants.SOCKET_LOGGER_NAME).removeHandler(self.capture_handler)
+        reset_memoize_cache()
```

Comment:
Can't we get away with just doing this at the end of each tearDown, not also during setUp?